src/Cms/User.php

```diff
@@ -244,6 +262,24 @@ public function email(): ?string
         return $this->email = $this->email ?? $this->credentials()['email'] ?? null;
     }
 
+    /**
+     * Encode a user email address
+     *
+     * @param string $email
+     * @return string
+     */
+    public static function encodeEmail(string $email)
```

Comment:
This should also be moved to the `Idn` class. It's no longer required for this bugfix, but because you have already implemented it, we might as well ship it for other uses.